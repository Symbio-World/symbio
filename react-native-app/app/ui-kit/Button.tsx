import * as React from 'react'
import { TouchableOpacity, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'

import { Icon, IconsEnum } from './Icon'

export enum ButtonStyle {
  FILL,
  OUTLINE,
  TRANSPARENT,
}

type Props = {
  onPress?: () => void
  title?: string
  icon?: IconsEnum
  variant?: ButtonStyle
}

const ButtonBaseStyle = {
  [ButtonStyle.FILL]: [t.bgPrimary],
  [ButtonStyle.OUTLINE]: [t.bgWhite, t.border2, t.borderPrimary],
  [ButtonStyle.TRANSPARENT]: [t.bgTransparent],
}

const InnerTitleBaseStyle = {
  [ButtonStyle.FILL]: [t.textWhite],
  [ButtonStyle.OUTLINE]: [t.textPrimary],
  [ButtonStyle.TRANSPARENT]: [t.textPrimary],
}

export const Button: React.FC<Props> = ({
  onPress = () => {},
  title = '',
  variant = ButtonStyle.FILL,
  icon = null,
}) => {
  const buttonStyle = ButtonBaseStyle[variant]
  const innerStyle = [InnerTitleBaseStyle[variant]]
  return (
    <TouchableOpacity
      style={[
        t.p4,
        t.itemsCenter,
        t.justifyCenter,
        t.flex,
        t.flexRow,
        t.rounded,
        ...buttonStyle,
      ]}
      onPress={onPress}
    >
      {icon ? <Icon style={[t.mR2]} iconType={icon} /> : null}
      <Text style={[t.textCenter, t.text2xl, t.textWhite, ...innerStyle]}>
        {title}
      </Text>
    </TouchableOpacity>
  )
}
