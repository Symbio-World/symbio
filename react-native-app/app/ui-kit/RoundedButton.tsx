import * as React from 'react'
import { TouchableOpacity, Text, View } from 'react-native'
import { t } from 'react-native-tailwindcss'

import { Icon, IconsEnum } from './Icon'
import { ButtonStyle } from './Button'

type Props = {
  onPress?: () => void
  innerTitle?: string
  outerTitle?: string
  icon?: IconsEnum
  buttonStyleType?: ButtonStyle
}

const ButtonBaseStyle = {
  [ButtonStyle.FILL]: [t.bgPrimary],
  [ButtonStyle.OUTLINE]: [t.bgWhite, t.border2, t.borderPrimary],
  [ButtonStyle.TRANSPARENT]: [t.bgTransparent],
}

const InnerTitleBaseStyle = {
  [ButtonStyle.FILL]: [t.textWhite],
  [ButtonStyle.OUTLINE]: [t.textPrimary],
  [ButtonStyle.TRANSPARENT]: [t.textPrimary],
}

export const RoundedButton: React.FC<Props> = ({
  onPress = () => {},
  innerTitle = '',
  outerTitle = '',
  buttonStyleType = ButtonStyle.FILL,
  icon = null,
}) => {
  const buttonStyle = ButtonBaseStyle[buttonStyleType]
  const innerStyle = InnerTitleBaseStyle[buttonStyleType]

  return (
    <TouchableOpacity onPress={onPress}>
      <View
        style={[
          ...buttonStyle,
          t.itemsCenter,
          t.justifyCenter,
          t.w20,
          t.h20,
          t.roundedFull,
        ]}
      >
        {innerTitle ? (
          <Text style={[t.textCenter, t.text2xl, ...innerStyle]}>
            {innerTitle}
          </Text>
        ) : null}
        {icon ? <Icon iconType={icon} /> : null}
      </View>
      <View>
        {outerTitle ? (
          <Text style={[t.textCenter, t.text2xl, t.textPrimary]}>
            {outerTitle}
          </Text>
        ) : null}
      </View>
    </TouchableOpacity>
  )
}
