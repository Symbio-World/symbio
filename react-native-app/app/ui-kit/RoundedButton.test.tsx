import * as React from 'react'
import { render } from 'react-native-testing-library'
import { RoundedButton } from './RoundedButton'
import { ButtonStyle } from './Button'

describe('RoundedButton', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<RoundedButton />)
    expect(toJSON()).toMatchSnapshot()
  })
  it('render transparent', () => {
    const { toJSON } = render(
      <RoundedButton buttonStyleType={ButtonStyle.TRANSPARENT} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })
  it('render fill', () => {
    const { toJSON } = render(
      <RoundedButton buttonStyleType={ButtonStyle.FILL} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })
  it('render outline', () => {
    const { toJSON } = render(
      <RoundedButton buttonStyleType={ButtonStyle.OUTLINE} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
