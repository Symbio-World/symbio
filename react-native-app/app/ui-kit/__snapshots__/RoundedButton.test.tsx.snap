// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedButton render fill 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#B35340",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "width": 64,
        },
        Object {
          "height": 64,
        },
        Object {
          "borderRadius": 9999,
        },
        Object {
          "padding": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": 20,
          },
          Object {
            "height": 20,
          },
        ]
      }
    />
  </View>
  <View />
</View>
`;

exports[`RoundedButton render inner title 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#B35340",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "width": 64,
        },
        Object {
          "height": 64,
        },
        Object {
          "borderRadius": 9999,
        },
        Object {
          "padding": 4,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "fontSize": 20,
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Button Test
    </Text>
    <View
      style={
        Array [
          Object {
            "width": 20,
          },
          Object {
            "height": 20,
          },
        ]
      }
    />
  </View>
  <View />
</View>
`;

exports[`RoundedButton render outer title 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#B35340",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "width": 64,
        },
        Object {
          "height": 64,
        },
        Object {
          "borderRadius": 9999,
        },
        Object {
          "padding": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": 20,
          },
          Object {
            "height": 20,
          },
        ]
      }
    />
  </View>
  <View>
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "fontSize": 20,
          },
          Object {
            "color": "#B35340",
          },
        ]
      }
    >
      Button Test
    </Text>
  </View>
</View>
`;

exports[`RoundedButton render outer title and icon 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#B35340",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "width": 64,
        },
        Object {
          "height": 64,
        },
        Object {
          "borderRadius": 9999,
        },
        Object {
          "padding": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": 20,
          },
          Object {
            "height": 20,
          },
        ]
      }
    >
      <SvgMock
        height="100%"
        width="100%"
      />
    </View>
  </View>
  <View>
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "fontSize": 20,
          },
          Object {
            "color": "#B35340",
          },
        ]
      }
    >
      Close
    </Text>
  </View>
</View>
`;

exports[`RoundedButton render outline 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "borderWidth": 2,
        },
        Object {
          "borderColor": "#B35340",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "width": 64,
        },
        Object {
          "height": 64,
        },
        Object {
          "borderRadius": 9999,
        },
        Object {
          "padding": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": 20,
          },
          Object {
            "height": 20,
          },
        ]
      }
    />
  </View>
  <View />
</View>
`;

exports[`RoundedButton render transparent 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "width": 64,
        },
        Object {
          "height": 64,
        },
        Object {
          "borderRadius": 9999,
        },
        Object {
          "padding": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": 20,
          },
          Object {
            "height": 20,
          },
        ]
      }
    />
  </View>
  <View />
</View>
`;

exports[`RoundedButton renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#B35340",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "width": 64,
        },
        Object {
          "height": 64,
        },
        Object {
          "borderRadius": 9999,
        },
        Object {
          "padding": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": 20,
          },
          Object {
            "height": 20,
          },
        ]
      }
    />
  </View>
  <View />
</View>
`;
